src/main/java/Duke.java

```diff
@@ -1,10 +1,205 @@
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Greeting.sayHello(isSinglish);
+
+        while (true) {
+            Scanner in = new Scanner(System.in);
+            String line = in.nextLine();
+
+            String[] commands = line.split(" ");
+            boolean isBye = commands[0].equals(BYE_COMMAND);
+            boolean isValidChangeLangCommand = commands[0].equals(CHANGE_COMMAND) && commands.length == 2 && commands[1].equals(LANG_COMMAND);
+            boolean isValidPrintListCommand = commands[0].equals(LIST_COMMAND) && commands.length == 1;
+            boolean isValidMarkOrUnmarkCommand = (commands[0].equals(MARK_COMMAND) || commands[0].equals(UNMARK_COMMAND)) && commands.length == 2;
+            boolean isTodo = commands[0].equals(TODO_COMMAND);
+            boolean isDeadline = commands[0].equals(DEADLINE_COMMAND);
+            boolean isEvent = commands[0].equals(EVENT_COMMAND);
+
+            if (isBye) {
+                Greeting.sayGoodbye(isSinglish);
+                break;
+            } else if (isValidChangeLangCommand) {
+                changeLanguage();
+            } else if (isValidPrintListCommand) {
+                printList();
+            } else if (isValidMarkOrUnmarkCommand) {
+                checkAndMarkTask(commands);
+            } else if (isTodo) {
+                addTodo(line);
+            } else if (isDeadline) {
+                addDeadline(line);
+            } else if (isEvent) {
+                addEvent(line);
+            } else {
+                Greeting.warnWrongSyntax(isSinglish);
+            }
+        }
+    }
+
+    /**
+     * Checks the validity of the event command.
+     * adds the event to the list if command has the right syntax.
+     * otherwise, it informs the user that it has a wrong syntax.
+     * @param line String entered by the user.
+     */
+    private static void addEvent(String line) {
+        int indexOfStartDate = line.indexOf(EVENT_START_FROM);
+        int indexOfEndDate = line.indexOf(EVENT_END_TO);
+        if (indexOfStartDate == OUT_OF_BOUNDS || indexOfEndDate == OUT_OF_BOUNDS) {
+            Greeting.warnWrongSyntax(isSinglish);
+        } else if (indexOfStartDate <= EVENT_COMMAND.length() + COMMAND_BUFFER) {
+            Greeting.warnEmptyDesc(isSinglish);
+        } else {
+            // +6 for "/from " length
+            String startOfDate = line.substring(indexOfStartDate + EVENT_START_FROM.length() + COMMAND_BUFFER, indexOfEndDate);
+            // +5 for "/to " length
+            String endOfDate = line.substring(indexOfEndDate + EVENT_END_TO.length() + COMMAND_BUFFER);
+            // 7 for "event " length
+            String descOfTask = line.substring(EVENT_COMMAND.length() + COMMAND_BUFFER, indexOfStartDate);
```

Comment:
Good use of comments and named constants to explain the use of the substring functions.